frontend/src/components/auth/Login.tsx

```diff
@@ -55,21 +53,29 @@ const Login = () => {
   };
 
   const onSignUpClick = async () => {
+    setInvalidLogin(false);
     if (isSignup) {
-      const user: AuthenticatedUser = await authAPIClient.signup(
-        firstName,
-        lastName,
-        email,
-        password,
-        signup,
-      );
-      setAuthenticatedUser(user);
+      if (agreeToTerms) {
+        const user: AuthenticatedUser = await authAPIClient.signup(
+          firstName,
+          lastName,
+          email,
+          password,
+          signup,
+        );
+        setAuthenticatedUser(user);
+      } else {
+        alert("Please agree to the terms and conditions in order to sign up.");
```

Comment:
nit: maybe a question for designers, but since this field is mandatory should we add an asterisk (*) at the end since we have it for the other mandatory fields?